## Case Study on Testing of Hypothesis
 A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and    decided to carry out a study on it.
 ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
 ● The company needs to check whether there is any dependency between thefeatures “Region” and “Manager”.
 Help the company to carry out their study with the help of data provided

In [13]:
# Importing required libraries

import numpy as np
import pandas as pd

In [14]:
# Loading the dataset

data = pd.read_csv(r"C:\Users\sravi\Downloads\ICTAK\statistics\Sales_add.csv")
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [15]:
# Basic information about the data

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1012.0+ bytes


##### The dataset has 22 rows and 5 columns. There is no missing value.

In [16]:
# Statistical analysis of the data

data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

##### Here we can see that the average sales value increased from 149239.954545 to 231123.727273 after the introduction of digital marketing. 

## Case 1 : To clarify whether there is any increase in sales after stepping into digital marketing.

Here the sample size is less than 30. So we perform "t test".

Hypothesis

* Null Hypothesis(H0): No increase in sales after applying digital marketing in company.

* Alternate Hypothesis(Ha): Sales increase after applying digital marketing in company.

* The Confidence level for this test will be 95% & the level of Significance, alpha = 0.05.


In [33]:
# Importing required library

import scipy.stats as stats
from scipy.stats import ttest_ind

In [34]:
# Now we will calculate test statistic and p-value 

t_stat,p_value = stats.ttest_ind(data["Sales_before_digital_add(in $)"],data["Sales_After_digital_add(in $)"])
print('Test statistic is:',t_stat)
print('p-value is:',p_value)

Test statistic is: -12.995084451110875
p-value is: 2.614368006904645e-16


In [35]:
# checking whether to accept or reject H0.

if p_value<0.05:
     print('Reject null hypothesies')
else:
    print('Accept null hypothesies')

Reject null hypothesies


Result- Therefore, there is an significant increase in sales after applying digital marketing in the company.



## Case 2 : to check whether there is any dependency between the features “Region” and “Manager”.


We have to check the dependency between the 'Region' and 'Manager'. So here we use 'Chi-square test of independence'.

Hypothesis

* Null Hypothesis(H0): There is no dependency between Region and Manager.
* Alternate Hypothesis(Ha): There is a dependency between Region and Manager.
* The Confidence level for this test will be 95% & the level of Significance, alpha = 0.05.   

In [39]:
# Extracting the Required Features, performing a crosstab on them and assigning it to a new variable
data_crosstab = pd.crosstab(data["Region"],data["Manager"])
data_crosstab

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [40]:
# Importing required library
from scipy.stats import chi2_contingency

# Now we will calculate Chi-square statistic, p-value, degree of freedom and expected frequency. 

chi_stat,p_value,dof,expected=chi2_contingency(data_crosstab)
print('Chi-square statistic is:',chi_stat)
print('p value is:',p_value)
print('Degree of freedom is:',dof)
print('Expected frequency is:',expected)

Chi-square statistic is: 3.050566893424036
p value is: 0.5493991051158094
Degree of freedom is: 4
Expected frequency is: [[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [41]:
# checking whether to accept or reject H0.

if p_value<0.05:
     print('Reject null hypothesies')
else:
    print('Accept null hypothesies')

Accept null hypothesies


 That is, there is no dependency between the features 'Region' and 'Manager'